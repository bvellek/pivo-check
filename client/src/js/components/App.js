import React, { Component } from 'react';
import '../../App.css';
import LandingHeader from './LandingHeader';
import LandingPage from './LandingPage';

class App extends Component {
  render() {
    return (
      <div className="App">
        <LandingHeader />
        <LandingPage />
      </div>
    );
  }
}

export default App;
