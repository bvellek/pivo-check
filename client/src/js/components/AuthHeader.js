import React from 'react';
import { Link } from 'react-router-dom';
import SVGLogo from './SVGLogo';

const AuthHeader = () => (
  <header className="auth-header top-header" role="banner">
    <Link to='/'>
      <h1>
        <span className="visually-hidden">Pivo-Check</span>
        <SVGLogo />
      </h1>
    </Link>
  </header>
);

export default AuthHeader;
