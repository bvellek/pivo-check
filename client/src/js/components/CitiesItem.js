import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import SVGDelete from './SVGDelete';

export class CitiesItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      deleteDialogStatus: false,
    };
  }

  deleteDialogChange = (e) => {
    e.preventDefault();
    this.setState({
      deleteDialogStatus: !this.state.deleteDialogStatus,
    });
  }

  deleteDialogClose = (e) => {
    e.preventDefault();
    this.setState({
      deleteDialogStatus: false,
    });
  }

  confirmDelete = (e) => {
    e.preventDefault();
    const cityDeleteID = this.props.cityInfo._id;
    this.props.dispatch(actions.deleteCity(cityDeleteID));
  }

  render() {
    const cityInfo = this.props.cityInfo;
    return (
      <div>
        <div className="city-item">
          <Link to={`/cities/${cityInfo._id}`}><h2 className="city-list-title">{cityInfo.cityName}</h2></Link>
          <div className="city-info">
            <p className="city-list-brew-count">{cityInfo.completedCount}/{cityInfo.brewTotal}</p>
            <div className="city-list-delete">
              <button type="submit" name="x-btn" onClick={this.deleteDialogChange}><SVGDelete /><span className="visually-hidden">Delete {cityInfo.cityName}</span></button>
            </div>
          </div>
        </div>

        <div className="delete-city-warning" hidden={!this.state.deleteDialogStatus}>
          <h3>Delete {cityInfo.cityName}?</h3>
          <p>Deleting a location will permanently remove it from your checklist and you will lose your visited brewery&nbsp;record.</p>
          <div className="delete-city">
            <button className="keep-city-btn" type="button" name="keep-btn" onClick={this.deleteDialogClose}>No, Keep {cityInfo.cityName}</button> {' '}
            <button className="delete-city-btn" type="button" name="remove-btn" onClick={this.confirmDelete}><SVGDelete />&nbsp;Yes, Delete {cityInfo.cityName}</button>
          </div>
        </div>
      </div>
    );
  }
}

export default connect()(CitiesItem);
