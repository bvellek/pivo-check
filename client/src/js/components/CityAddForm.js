import React, { Component } from 'react';
import PlacesAutocomplete, { geocodeByAddress } from 'react-places-autocomplete';

class CityAddForm extends Component {
  constructor(props) {
    super(props);
    this.state = { address: '' };
    this.onChange = (address) => this.setState({ address });
  }

  handleFormSubmit = (event) => {
    event.preventDefault();
    const { address } = this.state;

    geocodeByAddress(address, (err, { lat, lng }, results) => {
      if (err) { console.log('Oh no!', err); }

      console.log(`Yay! got latitude and longitude for ${address}`, { lat, lng }, 'whole results: ', results);
      console.log('new result: ', results[0].address_components[0].short_name, ', ', results[0].address_components[2].short_name);
    });
  }

  handleSelect = (address) => {
    this.setState({ address });

    geocodeByAddress(address, (err, { lat, lng }, results) => {
      if (err) { console.log('Oh no!', err); }

      console.log(`Yay! got latitude and longitude for ${address.locality}`, { lat, lng }, 'whole results: ', results);
      console.log('new result: ', results[0].address_components[0].short_name, ', ', results[0].address_components[2].short_name);
    });
  }

  render() {
    const options = {
      componentRestrictions: { country: 'US' },
      types: ['(cities)'],
    };

    const myStyles = {
      root: {
        position: 'relative',
        margin: '0 auto',
        width: '100%',
        maxWidth: '22em',
        padding: '0',
      },
      input: {
        boxSizing: 'border-box',
        width: '100%',
        lineHeight: '2em',
        margin: '0 auto',
        padding: '0 0.2em',
        color: '#44484D',
      },
      autocompleteItemActive: {
        backgroundColor: '#FFBAB3',
      },
    };

    const AutocompleteItem = ({ formattedSuggestion }) => (
      <div>
        <strong>{ formattedSuggestion.mainText }</strong>{' '}
        <small>{ formattedSuggestion.secondaryText }</small>
      </div>
    );

    return (

      <section className='city-add'>
        <form
          className='city-add-form'
          method='post'
          autoComplete='off'
          onSubmit={this.handleFormSubmit}
        >
          <div className='city-form-content'>
            <label htmlFor='city-search'>Add a city</label>
            <PlacesAutocomplete
              value={this.state.address}
              onChange={this.onChange}
              onSelect={this.handleSelect}
              options={options}
              inputId={'city-search'}
              inputName={'city'}
              styles={myStyles}
              autocompleteItem={AutocompleteItem}
              placeholder={'Seattle, WA'}
            />
            <button aria-label='Add city' type='submit' name='addCity'>Submit City 📍</button>
          </div>
        </form>
      </section>
    );
  }
}

export default CityAddForm;

