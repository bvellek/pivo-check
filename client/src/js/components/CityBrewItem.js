import React from 'react';

const CityBrewItem = ({ breweryInfo }) => (
  <li>
    <form className="checkoff-form" action="index.html" method="post">
      <label htmlFor={`check-${breweryInfo.id}`} className="visually-hidden">Checkoff {breweryInfo.brewery.name}</label>
      <div className='brew-check'>
        <input id={`check-${breweryInfo.id}`} type='checkbox' name={breweryInfo.id} />
          <div>&#10003;</div>
          <div>&#10003;</div>
          <div>&#10003;</div>
      </div>
    </form>
    <div className="brewery-info-container">
      <header>
        <h3>{breweryInfo.brewery.website ? (<a href={breweryInfo.brewery.website} target={`_${breweryInfo.id}`}>{breweryInfo.brewery.name}</a>) : (<a href={`https://www.google.com/search?q=${breweryInfo.brewery.name}`} target={`_${breweryInfo.id}`}>{breweryInfo.brewery.name}</a>)}</h3>
        {breweryInfo.brewery.images ? (<img className="brewery-logo" src={breweryInfo.brewery.images.squareMedium} alt={`${breweryInfo.brewery.name} Logo`} />) : (<img className="brewery-logo" src="https://placehold.it/200x200" alt="No Logo Provided" />)}

      </header>

      <form className="brewery-rating-form" action="index.html" method="post">
        <label htmlFor={`rating-${breweryInfo.id}`}>Rating: <span className="visually-hidden">for {breweryInfo.brewery.name}</span></label>
        <select id={`rating-${breweryInfo.id}`} className="brew-rating-select" name={`select-${breweryInfo.id}`}>
          <option defaultValue value="0">{' '}</option>
          <option value="1">üç∫</option>
          <option value="2">üç∫üç∫</option>
          <option value="3">üç∫üç∫üç∫</option>
        </select>
      </form>

        <div className="adr"><span>Address:</span>
          <a
            className="brewery-adr-link"
            href={`https://www.google.com/maps/place/${breweryInfo.streetAddress || ''},${breweryInfo.locality || ''},+${breweryInfo.region || ''}+${breweryInfo.postalCode || ''}`}
            target={`_map-${breweryInfo.id}`}
          >
          <div className="street-address">{breweryInfo.streetAddress}</div>
          <div className="extended-address">{breweryInfo.extendedAddress}</div>
          <span className="locality">{breweryInfo.locality}</span>
          <span className="comma-one">, </span>
          <span className="region">{breweryInfo.region}</span>{' '}
          <span className="postal-code">{breweryInfo.postalCode}</span>
          </a>
        </div>
        {breweryInfo.phone
          ? (<div className="brewery-phone">
            <span id={`phone-${breweryInfo.id}`}>Phone: </span>
            <a href={`tel:${breweryInfo.phone}`} className="tel" aria-labelledby={`phone-${breweryInfo.id}`}>{breweryInfo.phone}</a>
            </div>)
          : <div />
        }

      <details>
        <summary>Details <span className="visually-hidden">for {breweryInfo.brewery.name}</span></summary>
        <p><span>Type: </span>{breweryInfo.locationTypeDisplay}</p>
        <p className="brewery-description">{breweryInfo.brewery.description}</p>
      </details>
    </div>
  </li>
);

export default CityBrewItem;
