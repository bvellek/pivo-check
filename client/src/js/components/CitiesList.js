import React from 'react';
import CitiesItem from './CitiesItem';

const CitiesList = ({
  cities,
}) => (
  <section className="cities-list">
    <ul>
      {cities.map((city, index) => (
        <li key={index}><CitiesItem cityInfo={city} /></li>
      ))}
    </ul>
  </section>
);

export default CitiesList;
