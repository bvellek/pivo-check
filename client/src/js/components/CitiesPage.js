import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import InAppHeader from './InAppHeader';
import CityAddForm from './CityAddForm';
import SVGDelete from './SVGDelete';
import Footer from './Footer';

class CitiesPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  componentDidMount() {
    this.props.dispatch(actions.cleanAuth());
  }

  render() {
    return (
      <div>
        <InAppHeader />
        <div className="cities-page">
          <main role="main">
            <h1>My Cities</h1>
            <section className="city-add">
              <CityAddForm />
            </section>

          {/* if no cities added */}
            <section className="no-cities">
              <h2>Add a City!</h2>
              <p>Looks like you haven't added any cities to your checklist. Submit above to add a city full of breweries!</p>
            </section>

        {/* default */}
            <section className="cities-list">
              <ul>
                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">Seattle, WA</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">7/20</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="submit" name="button"><SVGDelete /><span className="visually-hidden">Delete Seattle</span></button>
                      </form>
                    </div>
                  </div>

                  <div className="delete-city-warning">
                    <h3>Delete Seattle?</h3>
                    <p>Deleting a location will permanently remove it from your checklist and you will lose your visited brewery record.</p>
                    <form className="delete-city-form" action="index.html" method="post">
                      <button className="keep-city-btn" type="button" name="button">No, Keep Seattle</button> {' '}
                      <button className="delete-city-btn" type="button" name="button"><SVGDelete />&nbsp;Yes, Delete Seattle</button>
                    </form>
                  </div>
                </li>

                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">Portland, OR</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">2/68</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="button" name="button"><SVGDelete /><span className="visually-hidden">Delete Portland</span></button>
                      </form>
                    </div>
                  </div>
                </li>
                <li>
                  <div className="city-item">
                    <a href="#"><h2 className="city-list-title">FL. Pops Hammock Seminole Village, FL</h2></a>
                    <div className="city-info">
                      <p className="city-list-brew-count">5/19</p>
                      <form className="city-list-delete-form" action="index.html" method="post">
                        <button type="button" name="button"><SVGDelete /><span className="visually-hidden">Delete Austin</span></button>
                      </form>
                    </div>
                  </div>
                </li>
              </ul>
            </section>
          </main>
          <Footer />
        </div>
      </div>
    );
  }
}

export default connect()(CitiesPage);

