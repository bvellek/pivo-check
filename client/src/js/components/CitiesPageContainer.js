import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import CitiesPage from './CitiesPage';
import CitiesList from './CitiesList';
import NoCities from './NoCities';

class CitiesPageContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
    };
  }

  componentDidMount() {
    document.querySelector('head > title').innerHTML = 'My Cities | PIVO-CHECK';
    this.props.dispatch(actions.cleanAuth());
    this.props.dispatch(actions.getCities());
  }

  addCity = (e) => {
    e.preventDefault();
  }

  // changeUser = (e) => {
  //   const field = e.target.name;
  //   const user = this.state.user;
  //   user[field] = e.target.value;

  //   this.setState({
  //     user,
  //   });
  // }

  // processForm = (e) => {
  //   e.preventDefault();
  //   this.props.dispatch(actions.loginUser(this.state.user));
  //   console.log(`email: ${this.state.user.email}`);
  //   console.log(`password: ${this.state.user.password}`);
  // }

  // demoSubmit = (e) => {
  //   e.preventDefault();
  //   this.setState({
  //     user: {
  //       password: 'demoPassword',
  //       email: 'demouser@PivoCheck.com',
  //     },
  //   }, () => {
  //     document.querySelector('.login-btn').click();
  //   });
  //   console.log(`email: ${this.state.user.email}`);
  //   console.log(`password: ${this.state.user.password}`);
  // }

  render() {
    const citiesArr = this.props.myCities;
    let citiesDisplay;
    if (citiesArr.length > 0) {
      citiesDisplay = <CitiesList cities={citiesArr} />;
    } else {
      citiesDisplay = <NoCities />;
    }

    return (
      <CitiesPage
        loadingStatus={this.props.loadingStatus}
        addCity={this.addCity}
        myCities={citiesDisplay}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  loadingStatus: state.cities.citiesLoadingStatus,
  myCities: state.cities.myCities,
});

export default connect(mapStateToProps)(CitiesPageContainer);


// <Route exact path="/login" render={() => ( // eslint-disable-line
//           !this.props.loginRedirect
//           ? (<LoginPage
//             onSubmit={this.processForm}
//             onChange={this.changeUser}
//             demoSubmit={this.demoSubmit}
//             user={this.state.user}
//             registrationSuccessMessage={this.props.registrationSuccessMessage}
//             errors={this.props.errors}
//           />)
//           : <Redirect to="/cities" />
//         )}
//       />
