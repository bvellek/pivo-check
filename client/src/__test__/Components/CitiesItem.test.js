import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { CitiesItem } from '../../js/components/CitiesItem';

describe('<CitiesItem />', () => {
  const callback = jest.fn();
  const city = {
    _id: '83jdi3dTEST',
    cityName: 'Test, TEST',
    brewTotal: 10,
  };
  const testComponent = (<CitiesItem
    deleteDialogChange={callback}
    deleteDialogClose={callback}
    confirmDelete={callback}
    cityInfo={city}
    registrationSuccessMessage={{}}
    errors={{}}
    loadingStatus={false}
  />);


  it('renders without crashing', () => {
    shallow(testComponent);
  });
  it('it renders correctly', () => {
    const component = shallow(testComponent);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
