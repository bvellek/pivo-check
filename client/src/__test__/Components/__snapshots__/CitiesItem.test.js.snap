exports[`<CitiesItem /> it renders correctly 1`] = `
<div>
  <div
    className="city-item">
    <Link
      replace={false}
      to="/cities/83jdi3dTEST">
      <h2
        className="city-list-title">
        Test, TEST
      </h2>
    </Link>
    <div
      className="city-info">
      <p
        className="city-list-brew-count">
        /
        10
      </p>
      <div
        className="city-list-delete">
        <button
          name="x-btn"
          onClick={[Function]}
          type="submit">
          <SVGDelete />
          <span
            className="visually-hidden">
            Delete 
            Test, TEST
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    className="delete-city-warning"
    hidden={true}>
    <h3>
      Delete 
      Test, TEST
      ?
    </h3>
    <p>
      Deleting a location will permanently remove it from your checklist and you will lose your visited brewery record.
    </p>
    <div
      className="delete-city">
      <button
        className="keep-city-btn"
        name="keep-btn"
        onClick={[Function]}
        type="button">
        No, Keep 
        Test, TEST
      </button>
       
       
      <button
        className="delete-city-btn"
        name="remove-btn"
        onClick={[Function]}
        type="button">
        <SVGDelete />
         Yes, Delete 
        Test, TEST
      </button>
    </div>
  </div>
</div>
`;
